samcli/lib/build/workflow_config.py

```diff
@@ -258,6 +254,47 @@ def _key(c: CONFIG) -> str:
     return True, None
 
 
+def supports_specified_workflow(specified_workflow: str) -> bool:
```

Comment:
I am having an hard time understand what this is actually for. Could you describe the use-case for this?

Note: This is probably just a gap in my understanding.

Reply:
Regarding [comment](https://github.com/aws/aws-sam-cli/pull/4665#discussion_r1099260481) to keep `_get_image` clean. If this method returns True, the `specified_workflow` (`BuildMethod` in `.yaml`) will overwrite `runtime` to get docker image. One concern is, `dotnet7` is not the only one that has `specified_workflow`, but we know for sure it is safe because container builds for `dotnet` wasn't supported. For other runtimes consuming `provided.al2`, it's hard to say it wouldn't break. So I think this check is necessary before passing down `specified_workflow`. Lmk if there's a better way to do this.